update_cache.py

```diff
@@ -64,10 +65,31 @@ def main():
         sys.exit(1)
 
     # let's grab sha and version info as Tuxsuite has the most up-to-date info
-    # we only need the first entry
-    first_entry = builds[next(iter(builds))]
-    git_sha = first_entry["git_sha"]
-    clang_version = first_entry["tuxmake_metadata"]["compiler"]["version_full"]
+    builds_that_are_missing_metadata = []
+    git_sha = None
+    clang_version = None
+    for entry, build in builds.items():
+        try:
+            git_sha = build["git_sha"]
+            clang_version = build["tuxmake_metadata"]["compiler"][
+                "version_full"]
+        except KeyError:
+            builds_that_are_missing_metadata.append(entry)
```

Comment:
```
    else:
        break
```

so that we don't keep iterating through `builds` once we have a hash and version?

Reply:
Yes.

[5069c26](https://github.com/ClangBuiltLinux/continuous-integration2/pull/683/commits/5069c260af95a8872a3e181ceb823428c2d5b705)